### PR TITLE
[FEATURE] Allow setting result format when calling batch.validate

### PR DESCRIPTION
**Pull request by tyler-hoffman**
See title.

### Other stuff
* Creates a single union type for all the things result format can be
* Adds a constant for the default result format

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/core/result_format.py
Comment (cdkini) [inline great_expectations/core/result_format.py:18]: Is this truly the default across all uses? I remember we had some inconsistencies

**Review by cdkini — APPROVED** [review]

File: great_expectations/core/result_format.py
Comment (tyler-hoffman) [reply great_expectations/core/result_format.py:18]: I believe it is now. The reason I made the constant was because we did have inconsistencies before, and that seemed back. I'm going to hold this PR open a bit more before merging, so feel free to push back if there's something I'm not thinking of.

**Pull request merged**
